MGDAPI-1620: Add creation of quota config map at operator startup

sergioifg94 opened the pull request:
# Description

> Link to JIRA: https://issues.redhat.com/browse/MGDAPI-1620

This is a temporal solution to include the quota configuration in the installation. Add the logic to create the quota ConfigMap at
operator startup, after the RHMI CR is created/discovered.

## Verification steps

Run the operator locally, ensuring that the quota ConfigMap doesn't exist in the cluster, and ensure that the installation progresses and finishes successfully

1. Checkout this PR
2. Prepare the cluster
    ```sh
    INSTALLATION_TYPE=managed-api make cluster/prepare/local
    ```
3. Ensure the ConfigMap doesn't exist
    ```sh
    oc delete cm/quota-config-managed-api-service -n redhat-rhoam-operator
    ```
4. Run the operator
    ```sh
    INSTALLATION_TYPE=managed-api make code/run
    ```
5. Verify that the ConfigMap is created and the installation progresses
    ```sh
    oc get cm/quota-config-managed-api-service -n redhat-rhoam-operator
    ```
    ```
    NAME                               DATA   AGE
    quota-config-managed-api-service   1      8s
    ```

## Type of change

<!-- Please delete options that are not relevant. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update <!-- Update JIRA with Affects -> Documentation -->
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [x] Verified independently on a cluster by reviewer

matskiv commented:
@sergioifg94 @laurafitzgerald Should we also remove the `config/configmap/quota-config-managed-api-service.yaml` and update Makefile?

sergioifg94 commented:
@matskiv I'd say it might be good to keep it for development purposes (e.g. to change the values in the ConfigMap using the make command), but it could still be possible to do it by changing the hardcoded value. Also this solution is only temporal so it might be good to keep the make command for when this is reverted

matskiv commented:
@sergioifg94 okay, but I think we need to remove implicit Configmap creation from the usual "cluster/prepare" flows that are used by all the tests. If we keep it, then we don't really know if it's created by make commands or new code added in this PR.

laurafitzgerald commented:
We should try to keep local install as close to managed tenants as possible. If we don't get config map from managed tenants we should remove it from make cluster prepare.

sergioifg94 commented:
@makslion @laurafitzgerald I just updated the PR, removed the ConfigMap creation from the make command, and also modified the ConfigMap creation code to use the variables `ConfigMapName` and `ConfigMapData` that were already defined

matskiv commented:
I verified that ConfigMap gets created by the operator and that the operator can still successfully switch to a different SKU.
/lgtm

@laurafitzgerald please merge this whenever suitable..

laurafitzgerald commented:
/hold
feature branch is under a rebase. Please do not merge this.

sergioifg94 commented:
Just rebased from the feature branch @laurafitzgerald @matskiv

matskiv commented:
/lgtm

laurafitzgerald commented:
/unhold

philbrookes commented:
/approve
